[rpc][profiler] add a test case to verify record_function context manager works
with RPC-based profiling

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34511 [rpc][profiler] add a test case to verify record_function context manager works
with RPC-based profiling**

With https://github.com/pytorch/pytorch/pull/34122/files, issues
with using record_function context manager and profiling RPCs were fixed. This
adds a test case to verify that we can use RPC with the `record_function`
decorator.

Differential Revision: [D20352242](https://our.internmc.facebook.com/intern/diff/D20352242/)

rohan-varma committed:
[rpc][profiler] add a test case to verify record_function context manager works
with RPC-based profiling

With https://github.com/pytorch/pytorch/pull/34122/files, issues
with using record_function context manager and profiling RPCs were fixed. This
adds a test case to verify that we can use RPC with the `record_function`
decorator.

Differential Revision: [D20352242](https://our.internmc.facebook.com/intern/diff/D20352242/)

[ghstack-poisoned]

mrshenli commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
Curious, besides the order, is there anything else we can get from the profiling result of "foo"?

rohan-varma commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
We should be guaranteed that the CPU time of foo_event_ix is greater than the RPC since it should be a superset. Let me look into adding this into the test.

rohan-varma committed:
Update on "[rpc][profiler] add a test case to verify record_function context manager works
with RPC-based profiling"

with RPC-based profiling**

With https://github.com/pytorch/pytorch/pull/34122/files, issues
with using record_function context manager and profiling RPCs were fixed. This
adds a test case to verify that we can use RPC with the `record_function`
decorator.

Differential Revision: [D20352242](https://our.internmc.facebook.com/intern/diff/D20352242/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc][profiler] add a test case to verify record_function context manager works
with RPC-based profiling"

with RPC-based profiling**
with RPC-based profiling**

With https://github.com/pytorch/pytorch/pull/34122/files, issues
with using record_function context manager and profiling RPCs were fixed. This
adds a test case to verify that we can use RPC with the `record_function`
decorator.

Differential Revision: [D20352242](https://our.internmc.facebook.com/intern/diff/D20352242/)

[ghstack-poisoned]
